core/src/test/resources/sql/cast-with-format.iq

```diff
@@ -517,173 +530,182 @@ select cast('2010 14 january' as date FORMAT
 !ok
 
-select cast(date'2010-07-19' as string FORMAT
+select cast(date'2010-07-19' as varchar FORMAT
     'MONTH Month month');
 EXPR$0
 JULY      July      july
 !ok
 
-select cast(date'2010-08-19' as string FORMAT
+select cast(date'2010-08-19' as varchar FORMAT
     'MONTH Month month');
 EXPR$0
 AUGUST    August    august
 !ok
 
-select cast(date'2010-09-19' as string FORMAT
+select cast(date'2010-09-19' as varchar FORMAT
     'MONTH Month month');
 EXPR$0
 SEPTEMBER September september
 !ok
+!}
 
+!if (false) {
+### disabled until Bug.CALCITE_6269_FIXED ###
 # Test odd casing of month token.
-select cast(date'2010-09-20' as string FORMAT
+select cast(date'2010-09-20' as varchar FORMAT
     'MOnth MONth MONTh');
 EXPR$0
 SEPTEMBER SEPTEMBER SEPTEMBER
 !ok
 
-select cast(date'2010-09-21' as string FORMAT
+select cast(date'2010-09-21' as varchar FORMAT
     'montH monTH moNTH moNTH');
 EXPR$0
 september september september september
 !ok
 
 # Test different lowercase vs uppercase scenarios with the datetime to string path
 # when FM is provided.
-select cast(date'2010-10-18' as string FORMAT
+select cast(date'2010-10-18' as varchar FORMAT
     'FMMONTH FMMonth FMmonth');
 EXPR$0
 OCTOBER October october
 !ok
 
-select cast(cast('2010-11-18' as timestamp) as string
+select cast(cast('2010-11-18' as timestamp) as varchar
     FORMAT 'FMMONTH FMMonth FMmonth');
 EXPR$0
 NOVEMBER November november
 !ok
 
-select cast(date'2010-12-19' as string FORMAT
+select cast(date'2010-12-19' as varchar FORMAT
     'FMMONTH FMMonth FMmonth');
 EXPR$0
 DECEMBER December december
 !ok
 
-select cast(date'2010-01-19' as string FORMAT
+select cast(date'2010-01-19' as varchar FORMAT
     'FMMONTH FMMonth FMmonth');
 EXPR$0
 JANUARY January january
 !ok
 
-select cast(date'2010-02-19' as string FORMAT
+select cast(date'2010-02-19' as varchar FORMAT
     'FMMONTH FMMonth FMmonth');
 EXPR$0
 FEBRUARY February february
 !ok
 
-select cast(date'2010-03-19' as string FORMAT
+select cast(date'2010-03-19' as varchar FORMAT
     'FMMONTH FMMonth FMmonth');
 EXPR$0
 MARCH March march
 !ok
 
-select cast(date'2010-04-19' as string FORMAT
+select cast(date'2010-04-19' as varchar FORMAT
     'FMMONTH FMMonth FMmonth');
 EXPR$0
 APRIL April april
 !ok
 
-select cast(date'2010-05-19' as string FORMAT
+select cast(date'2010-05-19' as varchar FORMAT
     'FMMONTH FMMonth FMmonth');
 EXPR$0
 MAY May may
 !ok
 
-select cast(date'2010-06-19' as string FORMAT
+select cast(date'2010-06-19' as varchar FORMAT
     'FMMONTH FMMonth FMmonth');
 EXPR$0
 JUNE June june
 !ok
 
-select cast(date'2010-07-19' as string FORMAT
+select cast(date'2010-07-19' as varchar FORMAT
     'FMMONTH FMMonth FMmonth');
 EXPR$0
 JULY July july
 !ok
 
-select cast(date'2010-08-19' as string FORMAT
+select cast(date'2010-08-19' as varchar FORMAT
     'FMMONTH FMMonth FMmonth');
 EXPR$0
 AUGUST August august
 !ok
 
-select cast(date'2010-09-19' as string FORMAT
+select cast(date'2010-09-19' as varchar FORMAT
```

Comment:
thanks for going through these im sure that took a bit

Reply:
Good to have the working tests enabled, and the disabled ones gives a status on whats yet to be supported and need to be addressed in followup JIRAs.